iati/core/tests/test_data.py

```diff
@@ -197,32 +212,43 @@ def test_instantiation_dataset_from_string_with_encoding(self, encoding):
     @pytest.mark.parametrize("encoding_declared, encoding_used", [
         ("UTF-16", "UTF-8"),
         ("UTF-16", "ISO-8859-1"),
+        ("UTF-16", "ASCII"),
         ("UTF-16", "BIG5"),
-        ("UTF-16", "EUC-JP"),
-        ("ASCII", "UTF-16"),
-        ("ISO-8859-1", "UTF-16"),
-        ("ISO-8859-2", "UTF-16"),
-        ("BIG5", "UTF-16"),
-        ("EUC-JP", "UTF-16")])
-    def test_instantiation_dataset_from_string_with_encoding_mismatch(self, encoding_declared, encoding_used):
+        ("UTF-16", "EUC-JP")
```

Comment:
I don't see any harm in testing encodings in the each direction. As such, I'd suggest using two `@pytest.mark.parametrize` statements (one for `encoding_declared`, one for `encoding_used`).

Also, a note that https://github.com/lxml/lxml/commit/e06b4b4f2d246d5c31d298a464948501b8a5dd91 adds `UTF-32` support, which can be added when that commit becomes part of an `lxml` release version.